### PR TITLE
miner, rpc: added submit hashrate for remote agents

### PR DESCRIPTION
**Pull request by obscuren**
This will allow you to query `eth_hashrate` to get a hash rate from any `ethminer` given that `ethminer` has implemented the `eth_submitHashrate` method.

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Thu Aug 6 09:58:49 UTC 2015_

File: miner/remote_agent.go [hunk 1]
```diff
@@ -63,7 +78,14 @@ func (a *RemoteAgent) Stop() {
...
 
-func (a *RemoteAgent) GetHashRate() int64 { return 0 }
+// GetHashRate returns the accumulated hashrate of all identifier combined
+func (a *RemoteAgent) GetHashRate() (tot int64) {
+	// this could overflow
+	for _, hashrate := range a.hashrate {
+		tot += int64(hashrate.rate)
+	}
+	return
+}
```
Comment (fjl) [inline miner/remote_agent.go]: Missing `a.hashrateMu.Lock`.

Comment (obscuren) [inline miner/remote_agent.go]: Thanks. I changed the mutex to `RW` thinking: "Don't forget to add the read lock" ... :-)

**Comment by fjl** [comment]
:+1: but maybe write a test.

**Comment by Gustav-Simonsson** [comment]
:+1:

**Pull request merged**